app/uk/gov/hmrc/exports/services/mapping/governmentagencygoodsitem/GovernmentAgencyGoodsItemBuilder.scala

```diff
@@ -63,20 +63,16 @@ class GovernmentAgencyGoodsItemBuilder @Inject()(
     exportItem: ExportItem,
     wcoGovernmentAgencyGoodsItem: WCOGovernmentAgencyGoodsItem,
     declarationType: DeclarationType
-  ) = {
+  ): Unit = {
     val combinedCommodity = for {
-      commodityWithoutGoodsMeasure <- mapItemTypeToCommodity(exportItem)
+      commodityWithoutGoodsMeasure <- mapExportItemToCommodity(exportItem)
       commodityOnlyGoodsMeasure = mapCommodityMeasureToCommodity(exportItem.commodityMeasure, declarationType)
     } yield combineCommodities(commodityWithoutGoodsMeasure, commodityOnlyGoodsMeasure)
 
     combinedCommodity.foreach(commodityBuilder.buildThenAdd(_, wcoGovernmentAgencyGoodsItem))
   }
-  private def mapItemTypeToCommodity(exportItem: ExportItem): Option[Commodity] =
-    (exportItem.itemType, exportItem.commodityDetails) match {
-      case (Some(item), Some(details)) =>
-        Some(cachingMappingHelper.commodityFromItemTypes(item, details, exportItem.dangerousGoodsCode, exportItem.cusCode, exportItem.taricCodes))
-      case _ => None
-    }
+  private def mapExportItemToCommodity(exportItem: ExportItem): Option[Commodity] =
```

Comment:
This method used to be:
```
  private def mapItemTypeToCommodity(itemType: Option[ItemType]): Option[Commodity] =
    itemType.map({ item =>
      cachingMappingHelper.commodityFromItemTypes(item)
    })
```

As `ItemType` is/was a required page, so the old method always returned Some(Commodity) which then had to be handled using the for comprehension.